packages/openapi-generator/src/generator-utils.ts

```diff
@@ -0,0 +1,50 @@
+import { promises } from 'fs';
+import { resolve } from 'path';
+import { ErrorWithCause } from '@sap-cloud-sdk/util';
+import {
+  GeneratorOptions,
+  ParsedGeneratorOptions,
+  generatorFlags
+} from './options';
+const { readFile, lstat } = promises;
+
+/**
+ * Parses a given path to a config file or directory and returns its content
+ * @param configPath path to a config file or a directory containing a config.json
+ * @returns Options to configure generation
+ */
+export async function parseOptionsFromConfig(
+  configPath: string
+): Promise<GeneratorOptions> {
+  try {
+    if ((await lstat(configPath)).isDirectory()) {
+      configPath = resolve(configPath, 'config.json');
+    }
+    return JSON.parse(await readFile(configPath, 'utf8'));
+  } catch (err) {
+    throw new ErrorWithCause(
+      `Could not read config.json at ${configPath}.`,
+      err
+    );
+  }
+}
+
+/**
+ * Removes default values from a given set of parsed generator options
+ * @param options parsed generator options
+ * @returns generator options with removed default values
+ */
+export function removeDefaultValues(
+  options: ParsedGeneratorOptions
+): GeneratorOptions {
+  return Object.entries(generatorFlags).reduce(
```

Comment:
I am not a fan of removing the values based based on the comparison of values.
If someone sets the default value on purpose it will be ignored.
Also, it is a bit counterintuitive and harder to read than the other approach with filtering the values based on the parsed 'raw' values.

Reply:
Default values are added after the `generate` function is executed regardless, but I agree that filtering with the raw values is the better approach.